app/backend/src/couchers/servicers/admin.py

```diff
@@ -584,6 +586,77 @@ def EditReply(self, request, context, session):
         obj.content = request.new_content.strip()
         return empty_pb2.Empty()
 
+    def GroupUsersAsDuplicated(self, request, context, session):
+        """Mark multiple users as duplicated accounts.
+        Users must belong to maximum one duplicate account group."""
+        req_users = request.users
+        if len(req_users) < 2:
+            context.abort(grpc.StatusCode.INVALID_ARGUMENT, errors.AT_LEAST_TWO_USERS_REQUIRED)
+        users = []
+        user_group_duplicated = None
+
+        for req_user in req_users:
+            user = session.execute(select(User).where_username_or_email_or_id(req_user)).scalar_one_or_none()
+            if not user:
+                context.abort(grpc.StatusCode.NOT_FOUND, errors.USER_NOT_FOUND)
+            users.append(user)
+
+            existing_user_group = user.get_group_duplicated()
+
+            # Check if user already belongs to a duplicate account group
+            if existing_user_group:
+                # Abort if other user is already in a different duplicate account group
+                if user_group_duplicated and user_group_duplicated.id != existing_user_group.id:
+                    context.abort(
+                        grpc.StatusCode.FAILED_PRECONDITION,
+                        errors.USER_ALREADY_BELONGS_TO_DUPLICATE_GROUP,
+                    )
+                user_group_duplicated = existing_user_group
+
+        # Create a new duplicate account group if no user already belongs to one
+        if not user_group_duplicated:
+            user_group_duplicated = UserGroup(group_type=UserGroupType.duplicate_account)
+            session.add(user_group_duplicated)
+            session.flush()
+
+        # Add users to the duplicate account group only if they are not already in it
+        for user in users:
+            if user not in user_group_duplicated.users:
+                user_group_duplicated.users.append(user)
+
+        session.commit()
```

Comment:
No need for a commit here: it autocommits when it leaves the `session_scope` contextmanager (defined in `interceptors.py:310` (see `SessionInterceptor`)